xchange-kraken/src/main/java/org/knowm/xchange/kraken/service/KrakenAccountService.java

```diff
@@ -38,9 +37,22 @@ public KrakenAccountService(Exchange exchange) {
   @Override
   public AccountInfo getAccountInfo() throws IOException {
 
+    KrakenTradeBalanceInfo krakenTradeBalanceInfo = getKrakenTradeBalance();
+    Wallet tradingWallet = KrakenAdapters.adaptWallet(getKrakenBalance());
+
+    Wallet marginWallet =
+        Wallet.Builder.from(tradingWallet.getBalances().values())
+            .id("margin")
+            .features(EnumSet.of(Wallet.WalletFeature.FUNDING, Wallet.WalletFeature.MARGIN_TRADING))
+            .maxLeverage(BigDecimal.valueOf(5))
+            .currentLeverage(
+                krakenTradeBalanceInfo
+                    .getCostBasis()
+                    .divide(krakenTradeBalanceInfo.getTradeBalance(), MathContext.DECIMAL32))
```

Comment:
just one thing to consider: can getTradeBalance return zero?
if so this will fails

Reply:
Eagle eye! I believe it is possible if you don't have any funds on the exchange. I will check for zero value and return zero.